federation/pkg/federation-controller/cluster/clustercontroller.go

```diff
@@ -180,11 +178,11 @@ func (cc *ClusterController) UpdateClusterStatus() error {
 		return err
 	}
 
-	cc.mu.Lock()
+	cc.Lock()
+	defer cc.Unlock()
```

Comment:
By default, `UpdateClusterStatus` gets called once every 40 seconds. Also the cluster additions/deletions are rare events and not critical path. so locking/unlocking once is better.